### PR TITLE
Fix samplTangent MKD coords, and test

### LINKED ISSUE #1574: Regression, new bug, related to useMsgLikelihoods
Looks like it has something to do with `getSolverParams(fg).useMsgLikelihoods = true`.  MWE:

```julia
using IncrementalInference

fg = generateGraph_CaesarRing1D()
getSolverParams(fg).useMsgLikelihoods = true

solveGraph!(fg)
```

Code completes fine if `useMsgLikelihoods=false`

_Originally posted by @dehann in https://github.com/JuliaRobotics/IncrementalInference.jl/issues/1572#issuecomment-1194719417_

### PR DESCRIPTION
**Pull request by Affie**
Pick some changes from #1575
Only changes I understood and knew would not break anything else. 
See .. diff: https://github.com/JuliaRobotics/IncrementalInference.jl/compare/22Q3/bug/1574_test..22Q3/bugfix/twig_1574

close #1574

### CODE REVIEW

File: src/ManifoldSampling.jl [hunk 1]
```diff
@@ -11,10 +11,11 @@ function sampleTangent end
...
 # Sampling MKD
 function sampleTangent(M::AbstractDecoratorManifold, x::ManifoldKernelDensity, p=mean(x))
-    # get legacy matrix of coordinates and selected labels
-    coords, lbls = sample(x.belief,1)
-    X = hat(x.manifold, p, coords)
-    return X
+  # get legacy matrix of coordinates and selected labels
+  #TODO make sure that when `sample` is replaced in MKD, coordinates is a vector
+  coords, lbls = sample(x.belief,1)
+  X = hat(x.manifold, p, coords[:])
```
Comment (Affie) [inline src/ManifoldSampling.jl:17]: `coords[:]` is the only change needed to fix the broken master.

Comment (Affie) [reply src/ManifoldSampling.jl:17]: The current sampling state is:
- sampling from beliefs (eg `sample` and `rand`) are in coordinates.
- sampleTangent is on the tangent at point `p` or the identity element for groups.

**Comment by Affie** [comment]
I think we should merge this and then look at the rest of #1575 again.

**Review by dehann — APPROVED** [review]

File: src/ManifoldSampling.jl (hunk 1 above)
Comment (dehann) [reply src/ManifoldSampling.jl:17]: See JuliaRobotics/Caesar.jl#895

**Pull request merged**